river/ensemble/boosting.py

```diff
@@ -119,3 +124,189 @@ def predict_proba_one(self, x):
+
+    [^2]: R. S. M. d. Barros, S. Garrido T. de Carvalho Santos and P. M. Gonçalves Júnior, "A Boosting-like Online Learning Ensemble," 2016 International Joint Conference on Neural Networks (IJCNN), 2016, pp. 1871-1878, doi: 10.1109/IJCNN.2016.7727427.
+    """
+
+    def __init__(self, model: base.Classifier, n_models=10, seed: int = None, error_bound=0.5):
+        super().__init__(model=model, n_models=n_models, seed=seed)
+        self.error_bound = error_bound
+        self.orderPosition = np.arange(n_models)
+        self.instances_seen = 0
+
+    def learn_one(self, x, y):
+        self.instances_seen += 1
+
+        acc = np.zeros(self.n_models)
+        for i in range(self.n_models):
+            acc[i] = (
+                self.correct_weight[self.orderPosition[i]]
+                + self.wrong_weight[self.orderPosition[i]]
+            )
+            if acc[i] != 0:
+                acc[i] = self.correct_weight[self.orderPosition[i]] / acc[i]
+
+        # sort models in descending order by correct_weight / correct_weight + error_weight (best case insertion sort)
+        for i in range(1, self.n_models):
+            key_position = self.orderPosition[i]
+            key_acc = acc[i]
+            j = i - 1
+            while (j >= 0) and (acc[j] < key_acc):
+                self.orderPosition[j + 1] = self.orderPosition[j]
+                acc[j + 1] = acc[j]
+                j -= 1
+            self.orderPosition[j + 1] = key_position
+            acc[j + 1] = key_acc
+
+        correct = False
+        maxAcc = 0
+        minAcc = self.n_models - 1
+        lambda_poisson = 1
+
+        models = list(enumerate(self))
+        for i in range(len(models)):
+            if correct:
+                pos = self.orderPosition[maxAcc]
+                maxAcc += 1
+            else:
+                pos = self.orderPosition[minAcc]
+                minAcc -= 1
+
+            for _ in range(poisson(lambda_poisson, self._rng)):
+                models[pos][1].learn_one(x, y)
+
+            if models[pos][1].predict_one(x) == y:
+                self.correct_weight[pos] += lambda_poisson
+                lambda_poisson *= (self.instances_seen) / (2 * self.correct_weight[pos])
+                correct = True
+            else:
+                self.wrong_weight[pos] += lambda_poisson
+                lambda_poisson *= (self.instances_seen) / (2 * self.wrong_weight[pos])
+                correct = False
+        return self
+
+    def predict_proba_one(self, x):
+        y_proba = collections.Counter()
+        for i, model in enumerate(self):
+            model_weight = 0.0
+            if self.correct_weight[i] > 0.0 and self.wrong_weight[i] > 0.0:
+                epsilon = self.wrong_weight[i] / (self.correct_weight[i] + self.wrong_weight[i])
+                if epsilon <= self.error_bound:
+                    beta_inv = (1 - epsilon) / epsilon
+                    model_weight = np.log(beta_inv)
```

Comment:
```suggestion
                    model_weight = math.log(beta_inv)
```

Reply:
Solved in last numpy removals and self.models[ind] commit.